### PR TITLE
General QoL and package updates

### PR DESCRIPTION
**Pull request by GAsplund**
# Key Features
- Update a lot of packages
  - Add compatibility with updated packages
  - Addresses some security issues
- Add a CODEOWNERS file
- Optimize the Dockerfile for faster sequential builds by placing most commonly edited stages further down
- Use Docker build cache between builds
- Run database migrations on each startup
- Compose file for running a local version of Gamma (mainly for testing with `INFO` keys)

# Known Issues
- The image size is bigger :(

### CODE REVIEW

File: .github/workflows/code_validation.yml [hunk 1]
```diff
@@ -1,45 +1,66 @@
-name: Linting
+name: Code validation
 
 on:
   push:
     branches: [main]
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+*
```
Comment (Oscariremma) [inline .github/workflows/code_validation.yml]: Do we really want to run code validation when tagging a release? It should have been run (and fixed) before you do a tag?

Comment (Oscariremma) [reply .github/workflows/code_validation.yml]: It's not explicitly wrong, just feels kind of weird?

Comment (GAsplund) [reply .github/workflows/code_validation.yml]: I think you're right, it doesn't make much sense to run validation on code that you are in some capacity saying is already valid.

**Review by Oscariremma — APPROVED** [review]
LGTM

**Pull request merged**